### ISSUE TITLE
append using 2D float array fails-  NullPointException: (array append)

### ISSUE THREAD

**Issue by archiartymusictects** [ccrma/chuck#202]
the simplified code below fails on chuck 1.4.1.0 (Windows 10).

float  boolBowDirList2 [0][0];
float  temp2 [0];
temp2<<0.5;
boolBowDirList2 <<temp2; // NullPointerException: (array append) on line[12]

it works fine if the array type is int but not for float

**Comment by gewang** [comment]
thanks for the report! reproduced on 1.5.1.3-dev (macOS) and a resolution has been pushed in f4a04c57943284b30a6a92fda86e0ed5be1e3a61 and dc4fd6f63e894fd508ca7d4d16bd2a0aca77a779 and scheduled to be included in 1.5.1.3. Please let us know if you encountered any further issues!
```
// 2D float array
float foo[0][0];
// float array
float bar[0];
// append to bar
bar << 1 << 2 << 3 << 5;
// append 1D array into 2D array
foo << bar;

// print contents
for( float x[] : foo )
{
    for( float y : x )
    {
        <<< y >>>;
    }
}
```

**Comment by gewang** [comment]
thanks for your patience; going to consider this issue closed for now (please re-open if issues persist with this); if you'd like, come hang out in the new [ChucK Community Discord server](https://discord.gg/ENr3nurrx8) and if you are on [chuck-1.5.x.x](https://chuck.stanford.edu/release/), we hope you are enjoying it!
